Index: FDroid_OTA_Packager/workspace/META-INF/com/google/android/update-binary
===================================================================
--- FDroid_OTA_Packager.orig/workspace/META-INF/com/google/android/update-binary
+++ FDroid_OTA_Packager/workspace/META-INF/com/google/android/update-binary
@@ -30,13 +30,37 @@ ADDOND=80-fdroid.sh
 SYSTEM="/system"
 
 mount_system() {
-  mount /system
-
-  # Modern devices use /system as root ("/")
   system_as_root=`getprop ro.build.system_root_image`
   if [ "$system_as_root" == "true" ]; then
-    SYSTEM="/system/system"
+    SYSTEM="/system_root"
+    if mount $SYSTEM; then
+      return 0
+    fi
+    active_slot=`getprop ro.boot.slot_suffix`
+    if [ ! -z "$active_slot" ]; then
+      block=/dev/block/bootdevice/by-name/system$active_slot
+    else
+      block=/dev/block/bootdevice/by-name/system
+    fi
+    mkdir -p $SYSTEM
+    if mount -o rw $block $SYSTEM ||
+       mount -o rw $block $SYSTEM -t ext4 ||
+       mount -o rw $block $SYSTEM -t f2fs; then
+      return 0
+    fi
+  else
+    SYSTEM="/system"
+    if mount $SYSTEM; then
+      return 0
+    fi
+    # Try to get the block from /etc/recovery.fstab
+    block=`cat /etc/recovery.fstab | cut -d '#' -f 1 | grep /system | grep -o '/dev/[^ ]*' | head -1`
+    if [ -n "$block" ] && mount $block $SYSTEM; then
+      return 0
+    fi
   fi
+
+  return 1
 }
 
 cd /tmp
@@ -44,6 +68,7 @@ mkdir fdroid
 cd fdroid
 unzip -o "$3"
 
+umount /system
 mount_system
 
 rm -rf "${SYSTEM}/app/FDroid"*
@@ -88,8 +113,10 @@ cp ${FDROID} ${FDROIDDIR}
 chmod 644 ${FDROIDDIR}/${FDROID}
 
 cp ${ADDOND} ${SYSTEM}/addon.d/
+chcon u:object_r:system_file:s0 ${SYSTEM}/addon.d/${ADDOND}
 
-umount /system
+umount ${SYSTEM}
+sync
 
 echo -n -e 'ui_print done\n' > /proc/self/fd/$2
 echo -n -e 'ui_print\n' > /proc/self/fd/$2
