Index: FDroid_OTA_Packager/workspace/META-INF/com/google/android/update-binary
===================================================================
--- FDroid_OTA_Packager.orig/workspace/META-INF/com/google/android/update-binary
+++ FDroid_OTA_Packager/workspace/META-INF/com/google/android/update-binary
@@ -28,15 +28,42 @@ PERM_DIR="etc/permissions"
 PERM=permissions_org.fdroid.fdroid.privileged.xml
 ADDOND=80-fdroid.sh
 SYSTEM="/system"
+TARGET="/system_not_defined"
 
 mount_system() {
-  mount /system
-
-  # Modern devices use /system as root ("/")
   system_as_root=`getprop ro.build.system_root_image`
   if [ "$system_as_root" == "true" ]; then
-    SYSTEM="/system/system"
+    TARGET="/system_root"
+    SYSTEM="$TARGET/system"
+    if mount $TARGET; then
+      return 0
+    fi
+    active_slot=`getprop ro.boot.slot_suffix`
+    if [ ! -z "$active_slot" ]; then
+      block=/dev/block/bootdevice/by-name/system$active_slot
+    else
+      block=/dev/block/bootdevice/by-name/system
+    fi
+    mkdir -p $TARGET
+    if mount -o rw $block $TARGET ||
+       mount -o rw $block $TARGET -t ext4 ||
+       mount -o rw $block $TARGET -t f2fs; then
+      return 0
+    fi
+  else
+    TARGET="/system"
+    SYSTEM="$TARGET"
+    if mount $TARGET; then
+      return 0
+    fi
+    # Try to get the block from /etc/recovery.fstab
+    block=`cat /etc/recovery.fstab | cut -d '#' -f 1 | grep /system | grep -o '/dev/[^ ]*' | head -1`
+    if [ -n "$block" ] && mount $block $TARGET; then
+      return 0
+    fi
   fi
+
+  return 1
 }
 
 cd /tmp
